src/test/scala/org/broadinstitute/hail/variant/GenotypeSuite.scala

```diff
@@ -103,15 +104,39 @@ class GenotypeSuite extends TestNGSuite {
 
   def triangleNumberOf(i: Int) = (i * i + i) / 2
 
-  @Test def gtIndexGtPairIsId(): Unit = {
-    forAll(Gen.choose(0,10000)) { (idx) =>
+  @Test def gtIndexGtPairIsId() {
+    forAll(Gen.choose(0, 10000)) { (idx) =>
       Genotype.gtIndex(Genotype.gtPair(idx)) == idx
     }.check()
   }
 
-  @Test def gtPairAndGtPairSqrtEqual(): Unit = {
-    forAll(Gen.choose(0,10000)) { (idx) =>
+  @Test def gtPairAndGtPairSqrtEqual() {
+    forAll(Gen.choose(0, 10000)) { (idx) =>
       Genotype.gtPair(idx) == Genotype.gtPairSqrt(idx)
     }.check()
   }
+
+  @Test def testGtFromLinear() {
+    val gen = for (nGenotype <- Gen.choose(2, 10).map(triangleNumberOf);
+      dosageGen = Gen.partition(nGenotype, 32768);
+      results <- Gen.buildableOfN[Array, Array[Int]](10000, dosageGen)) yield results
```

Comment:
This is unnecessarily complicated.  Don't generate an array of dosages and then check properties on them independently.